src/jpeg2000Compression.cpp

```diff
@@ -65,21 +95,25 @@ std::unique_ptr<uint8_t[]> Jpeg2000Compression::writeToMemory(
   int32_t* green = opjImage->comps[1].data;
   int32_t* blue = opjImage->comps[2].data;
 
-  for (unsigned int y = 0; y < height; y++) {
-    const uint8_t* pixel = reinterpret_cast<const uint8_t*>(buffer) + y * pitch;
-
-    for (unsigned int x = 0; x < width; x++) {
-      *red = pixel[0];
-      *green = pixel[1];
-      *blue = pixel[2];
-      pixel += 3;
-      red++;
-      green++;
-      blue++;
-    }
+  const uint8_t* pixel = reinterpret_cast<const uint8_t*>(buffer);
+  const unsigned int pixels_count = height*width*3;
+  for (unsigned int y = 0; y < pixels_count; y+=3) {
+    *red = pixel[y];
+    *green = pixel[y+1];
+    *blue = pixel[y+2];
+    red++;
```

Comment:
Sidenote: what is the purpose of these RGB ++'s?

Reply:
see above